feedsx/src/main/java/com/likeminds/feedsx/media/view/MediaPickerActivity.kt

```diff
@@ -63,24 +65,77 @@ class MediaPickerActivity : BaseAppCompatActivity() {
         checkStoragePermission()
     }
 
-    private fun checkStoragePermission() {
-        PermissionManager.performTaskWithPermission(
-            this,
-            settingsPermissionLauncher,
-            { startMediaPickerFragment() },
-            Permission.getStoragePermissionData(),
-            showInitialPopup = true,
-            showDeniedPopup = true,
-            permissionDeniedCallback = object : PermissionDeniedCallback {
-                override fun onDeny() {
-                    onBackPressed()
+    // setups up on back pressed callback
+    private fun setupOnBackPressedCallback() {
+        onBackPressedDispatcher.addCallback(this) {
+            when (val fragment = supportFragmentManager.currentFragment(R.id.nav_host)) {
+                is MediaPickerFolderFragment -> {
+                    finish()
+                }
+
+                is MediaPickerItemFragment -> {
+                    fragment.onBackPressedFromFragment()
                 }
 
-                override fun onCancel() {
-                    onBackPressed()
+                is MediaPickerDocumentFragment -> {
+                    if (fragment.onBackPressedFromFragment()) {
+                        finish()
+                    }
+                }
+
+                else -> {
+                    finish()
                 }
             }
```

Comment:
move this `onBackPressed` at the bottom of the fragment, to maintain the readability and show the lifecycle of the activity